chain/src/tests/util.rs

```diff
@@ -36,43 +40,64 @@ pub(crate) fn start_chain(
 fn create_cellbase(number: BlockNumber) -> Transaction {
     TransactionBuilder::default()
         .input(CellInput::new_cellbase_input(number))
-        .output(CellOutput::new(0, vec![], H256::zero(), None))
+        .output(CellOutput::new(
+            5000,
+            vec![],
+            create_script().type_hash(),
+            None,
+        ))
         .build()
 }
 
 pub(crate) fn gen_block(
     parent_header: &Header,
-    nonce: u64,
     difficulty: U256,
     commit_transactions: Vec<Transaction>,
+    proposal_transactions: Vec<Transaction>,
     uncles: Vec<UncleBlock>,
 ) -> Block {
     let number = parent_header.number() + 1;
     let cellbase = create_cellbase(number);
-    let header = HeaderBuilder::default()
+    let header_builder = HeaderBuilder::default()
         .parent_hash(parent_header.hash().clone())
         .timestamp(unix_time_as_millis())
         .number(number)
         .difficulty(difficulty)
-        .nonce(nonce)
-        .build();
+        .cellbase_id(cellbase.hash())
+        .nonce(rand::random());
```

Comment:
I oppose introducing randomness, may cause some involuntary bug.

Reply:
After review original nonce value usage here, I think it can be removed completely, pushed a new commit, please review again, thanks.